pkg/apis/core/validation/validation.go

```diff
@@ -4094,10 +4094,13 @@ func ValidateNode(node *core.Node) field.ErrorList {
 	// That said, if specified, we need to ensure they are valid.
 	allErrs = append(allErrs, ValidateNodeResources(node)...)
 
-	if len(node.Spec.PodCIDR) != 0 {
-		_, err := ValidateCIDR(node.Spec.PodCIDR)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "podCIDR"), node.Spec.PodCIDR, "not a valid CIDR"))
+	if len(node.Spec.PodCIDRs) != 0 {
+		// validate all cidrs in node.Spec.PodCIDR[]
+		for idx, value := range node.Spec.PodCIDRs {
+			_, err := ValidateCIDR(value)
```

Comment:
For stage 1 I thought we wanted to ensure just one per address family, so we should probably validate that for pod IPs and node ranges, for now.

Reply:
i think it is better to add this validation in `multi cidr ipam`, right?

Reply:
agreed, not everyone uses the in-cluster allocator.

Reply:
@vllry  
No. it was called out correctly. Let us do validation in api so folks who are using custom allocators will get the validation error.